things/service.go

```diff
@@ -549,6 +623,102 @@ func (svc service) Identify(ctx context.Context, key string) (string, error) {
 	return client.ID, nil
 }
 
+func (svc service) VerifyConnections(ctx context.Context, req *magistrala.VerifyConnectionsReq) (mgclients.ConnectionsPage, error) {
+	uniqueThings := getUniqueValues(req.GetThingsId())
+	uniqueChannels := getUniqueValues(req.GetGroupsId())
+
+	totalConnectionsCnt := len(uniqueChannels) * len(uniqueThings)
+	resultChan := make(chan mgclients.ConnectionStatus, totalConnectionsCnt)
+	errorChan := make(chan error, 1)
+
+	var wg sync.WaitGroup
+	wg.Add(totalConnectionsCnt)
+
+	for _, th := range uniqueThings {
+		for _, ch := range uniqueChannels {
+			go func(thing, channel string) {
+				defer wg.Done()
+				authReq := &magistrala.AuthorizeReq{
+					Subject:     channel,
+					SubjectType: auth.GroupType,
+					Permission:  auth.GroupRelation,
+					Object:      thing,
+					ObjectType:  auth.ThingType,
+				}
+
+				_, err := svc.auth.Authorize(ctx, authReq)
+				var status string
+				switch {
+				case err == nil:
+					status = connected
+				case errors.Contains(err, svcerr.ErrAuthorization):
+					status = disconnected
+				default:
+					select {
+					case errorChan <- errors.Wrap(svcerr.ErrMalformedEntity, err):
+					default:
+					}
+					return
+				}
+
+				resultChan <- mgclients.ConnectionStatus{
+					ThingId:   thing,
+					ChannelId: channel,
+					Status:    status,
+				}
+			}(th, ch)
```

Comment:
Instead of go routines , we can try to use https://pkg.go.dev/golang.org/x/sync/errgroup.
like we use ing cmd/<svc>/main.go